kpi/deployment_backends/kobocat_backend.py

```diff
@@ -306,6 +317,30 @@ def set_active(self, active):
             'backend_response': json_response,
         })
 
+    def set_asset_uid(self, force=False):
+        """
+        PATCH `kpi_asset_uid` field of survey.
+        It lets `kc` know which `Asset` is related to each `XForm`.
+        Useful when a form is created from the legacy upload form.
+
+        Store results in self.asset._deployment_data
+        """
+        not_sync = (self.backend_response.get('kpi_asset_uid', None) is None
+                    or force)
+        if not_sync:
+            url = self.external_to_internal_url(self.backend_response['url'])
+            payload = {
+                'kpi_asset_uid': self.asset.uid
+            }
+            json_response = self._kobocat_request('PATCH', url, data=payload)
+            is_set = json_response['kpi_asset_uid'] == self.asset.uid
+            assert is_set
+            self.store_data({
+                'backend_response': json_response,
+            })
+            return True
+        return False
```

Comment:
Maybe I was distracted, but the first time I read this, I thought `return False` was unreachable. It wasn't until I was reading the management command code that I realized I'd misunderstood.

I would (as I'm sure you'd expect) rewrite this to return early if `kpi_asset_uid` is already synced.